llvm/test/CodeGen/AArch64/hadd-combine.ll

```diff
@@ -879,6 +891,18 @@ define <8 x i16> @uhadd_fixedwidth_v4i32(<8 x i16> %a0, <8 x i16> %a1)  {
   ret <8 x i16> %res
 }
 
+define <8 x i16> @shadd_fixedwidth_v4i32(<8 x i16> %a0, <8 x i16> %a1)  {
```

Comment:
shadd_fixedwidth_v4i32 -> shadd_fixedwidth_v8i16 (its matching the shsub instruction for v8i16 types)